Drivers/Src/CAN.c

```diff
@@ -0,0 +1,98 @@
+#include "CAN.h"
+
+static void floatToBytes(float val, uint8_t* bytes_array, uint8_t bytes);
+
+/**
+ * @brief   Initializes the CAN system
+ * @param   None
+ * @return  None
+ */
+void CAN_Init(void) {
+    BSP_CAN_Init(CAN1);
+}
+
+/**
+ * @brief   Transmits data onto the CANbus
+ * @param   CAN bus line
+ * @param   id : CAN id of the message
+ * @param 	payload : the data that will be sent.
+ * @return  0 if data wasn't sent, otherwise it was sent.
+ */
+int CAN_Send(CAN_t bus,CANId_t id, CANPayload_t payload) {
+
+	// Not sure if these are the ids we want but I took them from BPS as placeholders
+	switch (id) {
+		case MC_BUS:
+		case VELOCITY:
+		case MC_PHASE_CURRENT:
+		case VOLTAGE_VEC:
+		case CURRENT_VEC:
+		case BACKEMF:
+		case TEMPERATURE:
+		case ODOMETER_AMPHOURS:
+		case CAR_STATE:
+			return BSP_CAN_WRITE(bus,id, &payload.data.b, payload.bytes)

```

Comment:
CAR_STATE can use the payload.data.b
Everything else should use the payload.data.d (64 bits) for the message